docs/fundamentals/networking/sockets/tcp-classes.md

```diff
@@ -78,10 +78,196 @@ The preceding C# code:
+
+This constructor accepts only three `AddressFamily` values, otherwise it will throw a <xref:System.ArgumentException>. Valid values are:
+
+- [AddressFamily.InterNetwork](xref:System.Net.Sockets.AddressFamily.InterNetwork): for IPv4 socket.
+- [AddressFamily.InterNetworkV6](xref:System.Net.Sockets.AddressFamily.InterNetworkV6): for IPv6 socket.
+- [AddressFamily.Unknown](xref:System.Net.Sockets.AddressFamily.Unknown): this will attempt to create a dual-stack socket, similarly to the default constructor.
+
+Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient(AddressFamily.InterNetwork);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+```
+
+#### The <xref:System.Net.Sockets.TcpClient.%23ctor(System.Net.IPEndPoint)?displayProperty=nameWithType> constructor
+
+Upon creating the socket, this constructor will also **bind** it to the provided **local** `IPEndPoint`. The <xref:System.Net.IPEndPoint.AddressFamily?displayProperty=nameWithType> property is used to determine the address family of the socket.
+
+Consider the following TCP client code:
+
+```csharp
+// Example IPEndPoint object
+var ep = new IPEndPoint(IPAddress.Parse("127.0.0.1"), 5001);
+var client = new TcpClient(ep);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+// Example IPEndPoint object
+var ep = new IPEndPoint(IPAddress.Parse("127.0.0.1"), 5001);
+var socket = new Socket(ep.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
+socket.Bind(ep);
+```
+
+#### The <xref:System.Net.Sockets.TcpClient.%23ctor(System.String,System.Int32)?displayProperty=nameWithType> constructor
+
+This constructor will attempt to create a dual-stack similarly to the default constructor and **connect** it to the **remote** endpoint defined by `hostname` and `port`.
+
+Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient("www.example.com", 80);
+```
+
+The preceding TCP client code is functionally equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+socket.Connect("www.example.com", 80);
+```
+
+### Connect to server
+
+All `Connect`, `ConnectAsync`, `BeginConnect` and `EndConnect` overloads in `TcpClient` are functionally equivalent to the corresponding `Socket` methods.
+
+Consider the following TCP client code:
+
+```csharp
+var client = new TcpClient();
+client.Connect("www.example.com", 80);
+```
+
+The above `TcpClient` code is equivalent to the following socket code:
+
+```csharp
+var socket = new Socket(SocketType.Stream, ProtocolType.Tcp);
+socket.Connect("www.example.com", 80);
+```
+
+### Create a server socket
+
+In `TcpListener` class, basically, constructors initializes underlying socket. One of them is `TcpListener(IPAddress localaddr, int port)`.
```

Comment:
This sounds redundant to me considering the code sample below. Not sure how to rephrase it to make it more meaningful. Maybe we should just remove this sentence?

Reply:
Well, we can but I didn't want to leave a blank there.